File: 26.RemoveDuplicatesfromSortedArray/26. Remove Duplicates from Sorted Array.py
from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        nums[0:len(dict.fromkeys(nums).keys())] = dict.fromkeys(nums).keys()
        return len(dict.fromkeys(nums).keys())


if __name__ == '__main__':
    c = Solution()
    print(c.removeDuplicates([1,1,2]))
    print(c.removeDuplicates([1,1,2,3,4,4]))
    print(c.removeDuplicates([1,1,2,5,6,7]))

File: 26.RemoveDuplicatesfromSortedArray/26.RemoveDuplicatesfromSortedArray.go
package main

import "fmt"

func removeDuplicates(nums []int) int {
	i := 0
	for j := range nums {
		if nums[i] != nums[j] {
			i += 1
			nums[i] = nums[j]
		}
	}
	return i + 1
}

func main() {
	test1 := []int{1, 1, 2}
	fmt.Println(removeDuplicates(test1))
}
